initrd/bin/kexec-boot

```diff
@@ -1,13 +1,89 @@
 #!/bin/sh
 # Launches kexec from saved configuration entries
-set -e -o pipefail
-. /tmp/config
-. /etc/functions
+
+usage() {
+	cat <<End-of-message
+Usage
+=====
+
+Launches kexec from saved configuration entries
+	
+$0 -b /boot -e 'kexec params|...|...' [<options>]
+	
+Options
+-------
+
+### -a <additional confiurations>
+
+Anything provided to the -a argument is appended to the kernel command line (usually comes from parsing the grub, syslinux, BLS files). Ex.
+
+	-a 'export CONFIG_BOOT_KERNEL_ADD="intel_iommu=on intel_iommu=igfx_off"'
```

Comment:
Happy to fix the description but I am not sure what needs to be changed.  

Lines 57 and 122 use these settings to adjust $cmdadd and then in lines 98 and 114 the $cmdline var is added to $kexeccmd -- which is a call to 'kexec'.   As I understand it, the options are passed to kexec to become inputs to the kernel in the boot process.  

Do you mean that the options from -a might be processed in heads in addition to being passed to the booting kernel?

Reply:
@tlaurion  line 124 (in my code) even says "Failed to add required kernel commands: $cmdadd".   Is the current description in the PR misleading?  If not, this adds a lot of information that does not exist currently which is helpful for troubleshooting in the heads environment.

Reply:
@jtmoree-github-com 

> Lines 57 and 122 use these settings to adjust $cmdadd and then in lines 98 and 114 the $cmdline var is added to $kexeccmd -- which is a call to 'kexec'. As I understand it, the options are passed to kexec to become inputs to the kernel in the boot process.

Exactly. They are modifiers. Those commands are called recursively inside of normal Heads codepaths, taking what is in BOARD config, resulting into /etc/config file, that can be overrided by /etc/config.user (and flashed back into the ROM, measured in PCR7, so the user can modify Heads configuration options on actual ROM and where /etc/config/user is merge with precedence to overrides over /etc/config


> Do you mean that the options from -a might be processed in heads in addition to being passed to the booting kernel?

if you call it directly from command line, -a and -r will remove just what you pass on command line. In Heads codepath, those removal and additions come from board config, /etc/config.user overrides by defaut. Doc should make it clear if doc is added for users wanting to read doc instead of code, while this documentation will need to be maintained if code change.

Reply:
I hear your concerns as:

1. The help I have written for -a and -r does not include the context in which kexec-boot exists--namely board configs.  It is only discussing what happens between kexec-boot and kexec
2. Help documentation has to be kept in sync with working code so we should scope help to relevant information only

Is this correct?